docker/image.ts

```diff
@@ -96,13 +96,13 @@ export class Image extends pulumi.ComponentResource {
                     if (!localImageName) {
                         localImageName = imageName;
                     }
-                    const id = await buildAndPushImageAsync(localImageName, build, imageName, this, async () => {
+                    const id = await buildAndPushImageAsync(localImageName, build, imageName, this, registry && (async () => {
```

Comment:
I believe there will be an issue a few lines above here when `registry` is undefined because the code tries to unwrap `Outputs` though this value - which can be `undefined` now.

I'll fix this up with a follow-up commit to generally simplify this.